internal/configurator/configurator.go

```diff
@@ -0,0 +1,96 @@
+/* Package Configurator is responsible for creating tasks from auditjobs mentioned in the config.yaml
+ */
+
+package configurator
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/maplelabs/github-audit/gitprovider"
+	"github.com/maplelabs/github-audit/input"
+	"github.com/maplelabs/github-audit/internal/task"
+	"github.com/maplelabs/github-audit/logger"
+	"github.com/maplelabs/github-audit/utils"
+)
+
+const (
+	// Default Scheduling interval for each audit job.
+	DEFAULTDURATION = time.Duration(5 * time.Minute)
+)
+
+var (
+	log logger.Logger
+	// ErrNoTaskConfigured if there is no audit job configured.
+	ErrNoTaskConfigured = errors.New("no audit task configured")
+)
+
+func init() {
+	log = logger.GetLogger()
+}
+
+// StartProcessing returns list of tasks with valid credentials which can be scheduled.
+func StartProcessing(config input.Config) ([]*task.Task, error) {
+	log.Info("starting creation of tasks for configured audit jobs")
+	tasks := make([]*task.Task, 0)
+	for _, aj := range config.AuditJobs {
+		task, err := createTask(aj, config.Targets)
+		if err != nil {
+			log.Errorf("error[%v] in creating task for audit job %v", err, aj.Name)
+			continue
```

Comment:
How are you retrying tasks if few tasks fails and few passes?

Reply:
nope , if a task creation fails , only adding logs to log file for user to review the task config